packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -386,9 +386,28 @@ class StandardMessageCodec implements MessageCodec<Object?> {
       }
     } else if (value is String) {
       buffer.putUint8(_valueString);
-      final Uint8List bytes = utf8.encoder.convert(value);
-      writeSize(buffer, bytes.length);
-      buffer.putUint8List(bytes);
+      final Uint8List asciiBytes = Uint8List(value.length);
+      Uint8List? utf8Bytes;
+      int utf8Offset = 0;
+      // Only do utf8 encoding if we encounter non-ascii characters.
+      for(int i = 0; i < value.length; ++i) {
```

Comment:
If we're going to do this ourselves we should have a wide variety of unit tests to ensure that we cover both ascii/utf8 sufficently to ensure that the data is not corrupted

Reply:
Do you feel like those are sufficient?

Reply:
The new code has complete test coverage.  Every line is exercised by a test.  I can't imagine another test or input that would exercise it differently.

Reply:
If you feel that is sufficient, then that is fine. I also don't think you need a test exemption since you updated the benchmark, right?